src/Admin/SettingsPage/ConversionImport.php

```diff
@@ -0,0 +1,145 @@
+
+        $file_temp_name = $file['tmp_name'];
+        $file_name = $file['name'];
+        $file_type = $file['type'];
+
+        $target_dir = wp_get_upload_dir();
+        $new_file_path = $target_dir['path'].'/'.rand(1, 9999).'-'.$file_name;
+
+        if(!empty($file_name) && in_array($file_type, $csv_mines)) {
+            if(is_uploaded_file($file_temp_name) && move_uploaded_file($file_temp_name, $new_file_path)) {
+                //save csv
+                if(update_option($this->conversion_data, $new_file_path)) {
+                    wp_safe_redirect(add_query_arg('step', '2', MAILOPTIN_LEAD_IMPORT_CSV_SETTINGS_PAGE));
+                    exit;
+                } else {
+                    return __('There was an error, couldn\'t save the csv file path', 'mailoptin');
+                }
+            } else {
+                return __('CSV file could not be moved.', 'mailoptin');
+            }
+        } else {
+            return __('The uploaded file is not a csv file. Please try again.', 'mailoptin');
+        }
+    }
+
+    /**
+     * call to import function
+     *
+     */
+    public function import($fields) {
+        $csv_headers = $this->read_csv_headers();
+        $file_path = get_option($this->conversion_data);
+
+        $reader = Reader::createFromPath($file_path, 'r');
+
+        $data = $reader->fetchAssoc($csv_headers);
+        array_shift($data);
+
+        $insert_data = [];
+        $conversionRepoResponse = false;
+
+        foreach ($data as $key => $value) {
+            foreach ($fields as $field_key => $field_value) {
+                if (isset($value[$field_value])) {
+                    $insert_data[$field_key] = esc_html($value[$field_value]);
+                }
+            }
+
+            //add fields to data before passing
+
+            $insert_data['optin_campaign_id'] = 0; // since it's non mailoptin form, set it to zero.
+            $insert_data['optin_campaign_type'] = esc_html__('Import Leads with CSV', 'mailoptin');;
+            $insert_data['user_agent'] = esc_html($_SERVER['HTTP_USER_AGENT']);
+            $insert_data['conversion_page'] = __('Leads Import', 'mailoptin');
+            $insert_data['referrer'] = __('Leads Import', 'mailoptin');
+            $insert_data['custom_fields'] = NULL;
+
+            //get the conversion by email
+            $conversion_by_email = OptinConversionsRepository::get_conversions_by_email($insert_data['email']);
+            if($conversion_by_email) {
+                $conversionRepoResponse = OptinConversionsRepository::update($conversion_by_email[0]['id'], $insert_data);
+            } else {
+                //insert into repository
+                $conversionRepoResponse = OptinConversionsRepository::add($insert_data);
+            }
+        }
+
+
+        //If the conversion is true
+        if($conversionRepoResponse) {
```

Comment:
Importing not working. I mapped the fields to import, got "There was an error importing leads into the leadbank". I mapped only subscriber email and got the error below and the PHP notices.


![image](https://user-images.githubusercontent.com/4140659/106507915-fd8dcf00-64cb-11eb-9d87-66a7159b8ec4.png)

Reply:
This is the file I am trying to import https://share.getcloudapp.com/NQuJbDZj